cmd/util/cmd/epoch-recovery/cmd/reset.go

```diff
@@ -0,0 +1,192 @@
+package cmd
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+
+	"github.com/onflow/cadence"
+
+	jsoncdc "github.com/onflow/cadence/encoding/json"
+
+	"github.com/onflow/flow-go/engine/common/rpc/convert"
+	"github.com/onflow/flow-go/model/bootstrap"
+	"github.com/onflow/flow-go/state/protocol/inmem"
+	"github.com/onflow/flow-go/utils/io"
+)
+
+var (
+	flagBootDir string
+	flagPayout  uint64
+)
+
+// resetCmd represents a command to reset epoch data in the Epoch smart contract
+var resetCmd = &cobra.Command{
+	Use:   "reset",
+	Short: "Generates `resetEpoch` JSON transaction arguments.",
+	Long:  "Generates `resetEpoch` transaction arguments from a root protocol state snapshot and writes it to a JSON file",
+	Run:   resetRun,
+}
+
+func init() {
+	rootCmd.AddCommand(resetCmd)
+	addResetCmdFlags()
+}
+
+func addResetCmdFlags() {
+	resetCmd.Flags().StringVar(&flagBootDir, "boot-dir", "", "path to the directory containing the bootstrap files")
+	_ = resetCmd.MarkFlagRequired("boot-dir")
+
+	resetCmd.Flags().Uint64Var(&flagPayout, "payout", 0, "the payout")
+}
+
+// resetRun resets epoch data in the Epoch smart contract with fields generated
+// from the root-protocol-snapshot.json
+func resetRun(cmd *cobra.Command, args []string) {
+
+	// path to the JSON encoded cadence arguments for the `resetEpoch` transaction
+	path, err := os.Getwd()
+	if err != nil {
+		log.Fatal().Err(err).Msgf("could not get working directory path")
+	}
+	argsPath := filepath.Join(path, "reset-epoch-args.json")
+
+	// path to the root protocol snapshot json file
+	snapshotPath := filepath.Join(flagBootDir, bootstrap.PathRootProtocolStateSnapshot)
+
+	// check if root-protocol-snapshot.json file exists under the dir provided
+	exists, err := pathExists(snapshotPath)
+	if err != nil {
+		log.Fatal().Err(err).Str("path", snapshotPath).Msgf("could not check if root protocol-snapshot.json exists")
+	}
+	if !exists {
+		log.Fatal().Str("path", snapshotPath).Msgf("root-protocol-snapshot.json file does not exists in the `boot-dir` given")
+	}
+
+	// read root protocol-snapshot.json
+	bz, err := io.ReadFile(snapshotPath)
+	if err != nil {
+		log.Fatal().Err(err).Msgf("could not read root snapshot file")
+	}
+
+	// unmarshal bytes to inmem protocol snapshot
+	snapshot, err := convert.BytesToInmemSnapshot(bz)
+	if err != nil {
+		log.Fatal().Err(err).Msg("could not convert array of bytes to snapshot")
+	}
+
+	// extract arguments from reset epoch tx from snapshot
+	txArgs := extractResetEpochArgs(snapshot)
+
+	// ancode to JSON
+	enc := encodeArgs(txArgs)
+
+	// write JSON args to file
+	err = io.WriteFile(argsPath, enc)
+	if err != nil {
+		log.Fatal().Err(err).Msg("could not write jsoncdc encoded arguments")
+	}
```

Comment:
Let's include a success log here including the path of the new file. Better to be verbose with these tools